### PR TITLE
Feat/algo transaction builder factory

### PR DESCRIPTION
**Pull request by nimraarif21**
@ryan-bitgo @fdmirza @gibrangul95

### CODE REVIEW

**Comment by gibrangul95** [comment]
@ryan-bitgo small commit, all the conflicts have been resolved

File: modules/account-lib/src/coin/algo/transaction.ts [hunk 1]
```diff
@@ -99,17 +99,27 @@ export class Transaction extends BaseTransaction {
     const msigAddress = algosdk.multisigAddress(multiSigOptions);
     this._algoTransaction.from = algosdk.decodeAddress(msigAddress);
 
-    // Check if it is a signed or unsigned tx.
-    // If unsigned, sign it using first keypair and then append next signatures.
-    // If signed, appending next signatures.
-    let signedTx = this._signedTransaction
```
Comment (ryan-bitgo) [inline modules/account-lib/src/coin/algo/transaction.ts:105]: What was wrong with this? Seems to be a undo from a merge conflict.

**Review by ryan-bitgo — CHANGES_REQUESTED** [review]

**Review by tylerlevine — CHANGES_REQUESTED** [review]
Please include Jira ticket in both the commit message and PR body. Also could you please fix the double scope in the commit message?

File: modules/account-lib/src/coin/algo/transaction.ts (hunk 1 above)
Comment (nimraarif21) [reply modules/account-lib/src/coin/algo/transaction.ts]: I have changed this now. Please check

**Comment by nimraarif21** [comment]
> Please include Jira ticket in both the commit message and PR body. Also could you please fix the double scope in the commit message?

I do not have access to Jira.

**Pull request closed without merging**